generators/client/generator.ts

```diff
@@ -107,13 +107,16 @@ export default class JHipsterClientGenerator extends BaseApplicationGenerator {
   get composing() {
     return this.asComposingTaskGroup({
       async composing() {
-        const { clientFramework, testFrameworks } = this.jhipsterConfigWithDefaults;
+        const { clientFramework, testFrameworks, skipClient } = this.jhipsterConfigWithDefaults;
         if ([ANGULAR, VUE, REACT].includes(clientFramework)) {
           await this.composeWithJHipster(clientFramework);
         }
         if (Array.isArray(testFrameworks) && testFrameworks.includes(CYPRESS)) {
           await this.composeWithJHipster(GENERATOR_CYPRESS);
         }
+        if (!skipClient && clientFramework !== 'no') {
+          await this.composeWithJHipster('jhipster:java:node');
+        }
```

Comment:
client generator is designed to don't depend on server, this will force blueprints like nodejs or skipServer to add java based npmw.
This should be moved to spring-boot generator at composingComponent priority.